modules/cloud_logs/main.tf

```diff
@@ -106,3 +107,55 @@ resource "ibm_logs_outgoing_webhook" "en_integration" {
     region_id                       = each.value.en_region
   }
 }
+
+##############################################################################
+# Logs Routing
+##############################################################################
+resource "ibm_iam_authorization_policy" "logs_routing_policy" {
+  count               = local.create_access_policy_logs_routing ? 1 : 0
+  source_service_name = "logs-router"
+  roles               = ["Sender"]
+  description         = "Allow Logs Routing `Sender` access to the IBM Cloud Logs with ID ${ibm_resource_instance.cloud_logs.guid}."
+
+  resource_attributes {
+    name     = "serviceName"
+    operator = "stringEquals"
+    value    = "logs"
+  }
+
+  resource_attributes {
+    name     = "accountId"
+    operator = "stringEquals"
+    value    = data.ibm_iam_account_settings.iam_account_settings.account_id
+  }
+
+  resource_attributes {
+    name     = "serviceInstance"
+    operator = "stringEquals"
+    value    = ibm_resource_instance.cloud_logs.guid
+  }
+}
+
+##############################################################################
+# Random Suffix
+##############################################################################
+
+resource "random_string" "random_tenant_suffix" {
+  length  = 13
+  numeric = true
+  upper   = false
+  lower   = false
+  special = false
+}
+
+resource "ibm_logs_router_tenant" "logs_router_tenant_instance" {
+  name = var.logs_routing_tenant_name != null ? var.logs_routing_tenant_name : "${var.region}-${random_string.random_tenant_suffix.result}"
+  targets {
+    log_sink_crn = ibm_resource_instance.cloud_logs.crn
+    name         = local.instance_name
+    parameters {
+      host = "${ibm_resource_instance.cloud_logs.guid}.ingress.${var.region}.logs.cloud.ibm.com"
```

Comment:
Is this not a public host? We probably want to use private by default, but I guess we should also optionally allow user to public incase its not a VRF enabled account.

Reply:
For this, I used service_endpoints as reference, if Cloud logs is setup with public endpoint then use public endpoint or else private. 
Let me know if any concern.